src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -5408,7 +5408,20 @@ namespace winrt::TerminalApp::implementation
                 appPrivate->PrepareForAIChat();
             }
         }
-        _extensionPalette = winrt::Microsoft::Terminal::Query::Extension::ExtensionPalette();
+
+        // since we only support one type of llmProvider for now, just instantiate that one (the AzureLLMProvider)
+        // in the future, we would need to query the settings here for which LLMProvider to use
+        _lmProvider = winrt::Microsoft::Terminal::Query::Extension::AzureLLMProvider();
+        auto setAuthenticationValues = [&]() {
```

Comment:
pretty sure that capturing `this` via `&` is considered harmful at this point. Pretty sure we've got to do the `weak_this` thing we do everywhere else (and/or promote `setAuthenticationValues` to be a method on `TerminalPage`)

Reply:
Because the lambda isn't a coroutine (async lifetime), and since we protect its "sync" lifetime with the `_azureOpenAISettingChangedRevoker`, taking the raw `this` pointer in is safe.

But I agree that making this a method would be better, because that makes it slightly more readable and also prevents accidental bugs. For instance, the `[&]` the could cause you to accidentally refer to a object around the closure (like a string or whatever) which goes out of scope at the end of the function.